OpenMEEG/libs/OpenMEEG/src/assembleSourceMat.cpp

```diff
@@ -118,27 +118,26 @@ namespace OpenMEEG {
                 domain = geo.domain(domain_name);
             }
             const double sigma = domain.sigma();
-	    //only consider dipoles in non-zero conductivity domain
-	    if(!sigma==0)
```

Comment:
This cannot be the proper fix.... if (sigma!=0.0) ???
I do not know the clang warning, but this seems rather valid code (even if not the most readable). I Can see two possible complaints : The ! == or the fact that 0 is an integer constant and not a double one. The above proposal fixes both.... I have no access to your branch, otherwise I would have done the change myself.